src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -212,29 +190,112 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
+                    services.TryAddExisting<IGrainFactory, GrainFactory>();
+                    services.TryAddExisting<IInternalGrainFactory, GrainFactory>();
+                    services.TryAddSingleton<TypeMetadataCache>();
+                    services.TryAddSingleton<AssemblyProcessor>();
+                    services.TryAddSingleton<ActivationDirectory>();
+                    services.TryAddSingleton<LocalGrainDirectory>();
+                    services.TryAddExisting<ILocalGrainDirectory, LocalGrainDirectory>();
+                    services.TryAddSingleton<SiloStatisticsManager>();
+                    services.TryAddSingleton<ISiloPerformanceMetrics>(sp => sp.GetRequiredService<SiloStatisticsManager>().MetricsTable);
+                    services.TryAddSingleton<SiloAssemblyLoader>();
+                    services.TryAddSingleton<GrainTypeManager>();
+                    services.TryAddExisting<IMessagingConfiguration, GlobalConfiguration>();
+                    services.TryAddSingleton<MessageCenter>();
+                    services.TryAddExisting<IMessageCenter, MessageCenter>();
+                    services.TryAddExisting<ISiloMessageCenter, MessageCenter>();
+                    services.TryAddSingleton<Catalog>();
+                    services.TryAddSingleton(sp => sp.GetRequiredService<Catalog>().Dispatcher);
+                    services.TryAddSingleton<InsideRuntimeClient>();
+                    services.TryAddExisting<IRuntimeClient, InsideRuntimeClient>();
+                    services.TryAddSingleton<MembershipFactory>();
+                    services.TryAddSingleton<MultiClusterOracleFactory>();
+                    services.TryAddSingleton<LocalReminderServiceFactory>();
+                    services.TryAddSingleton<DeploymentLoadPublisher>();
+                    services.TryAddExisting<IDeploymentLoadPublisher, DeploymentLoadPublisher>();
+                    services.TryAddSingleton<IMembershipTable>(
+                        sp => sp.GetRequiredService<MembershipFactory>().GetMembershipTable(sp.GetRequiredService<GlobalConfiguration>()));
+                    services.TryAddSingleton<MembershipOracle>(
+                        sp =>
+                        sp.GetRequiredService<MembershipFactory>()
+                          .CreateMembershipOracle(sp.GetRequiredService<Silo>(), sp.GetRequiredService<IMembershipTable>()));
+                    services.TryAddExisting<IMembershipOracle, MembershipOracle>();
+                    services.TryAddExisting<ISiloStatusOracle, MembershipOracle>();
+                    services.TryAddSingleton<Func<ISiloStatusOracle>>(sp => () => sp.GetRequiredService<ISiloStatusOracle>());
+                    services.TryAddSingleton<MultiClusterOracleFactory>();
+                    services.TryAddSingleton<LocalReminderServiceFactory>();
+                    services.TryAddSingleton<SiloControl>();
```

Comment:
hmmm, registering this and other system targets in the container can be extremely dangerous, since callers should never resolve them as is, and instead should get a reference to the system target via the internal grain factory.
Maybe instead of registering it here (which I assume it's only because you want constructor parameters to be injected at construction time), you can use `ActivatorUtilities.CreateInstance` when registering it as a system target, but never register it as a service.
The thing that would be legal to register in the container is `ISiloControl` which can ask the IInternalGrainFactory to get that system target, but not these concrete objects that should never be used from the container directly.